edx_when/api.py

```diff
@@ -247,13 +261,17 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
 
     with transaction.atomic():
```

Comment:
Can you make an optional param that disables calling this atomic() wrapper? That way we can just do everything in one transaction when we're updating all the course instead of 10K smaller nested transactions?

Reply:
Are sub-transactions (django calls them snapshots) expensive? We can add `snapshot=False` to `atomic` to disable snapshots, if we don't like them here. I'm not clear on the tradeoffs.

Reply:
I suspect, but I'm not sure. I do think that there's no value in having individual save points when doing a bulk update, because we don't want to half-update a course.

Reply:
I've updated this PR with `savepoint=False` on the block-level method. I kept it on the course-level method, because that's less likely to be called in a loop? And whatever benefit we get from savepoints might be useful there?